Competition2017/src/competition/subsystems/drive/commands/FieldOrientedTankDriveWithJoystick.java

```diff
@@ -0,0 +1,69 @@
+package competition.subsystems.drive.commands;
+
+import java.util.function.DoubleSupplier;
+
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import xbot.common.controls.sensors.XJoystick;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.XPropertyManager;
+
+public class FieldOrientedTankDriveWithJoystick extends BaseDriveCommand {
+
+    private final PoseSubsystem poseSubsystem;
+    
+    private final XJoystick leftJoystick;
+    
+    private final PIDManager headingDrivePid;
+    public final double defaultPValue = 1 / 80d;
+ 
+    private ContiguousHeading targetHeading;
+    protected DoubleSupplier targetHeadingSupplier;
+    
+    public FieldOrientedTankDriveWithJoystick(
+            OperatorInterface oi, 
+            DriveSubsystem driveSubsystem, 
+            PoseSubsystem poseSubsystem,
+            XPropertyManager propManager,
+            RobotAssertionManager assertionManager) {
+        super(driveSubsystem);
+        this.poseSubsystem = poseSubsystem;
+        this.leftJoystick = oi.leftJoystick;
+        headingDrivePid = new PIDManager("Rotate to heading", propManager, assertionManager, defaultPValue, 0, 0);
+        
+        headingDrivePid.setErrorThreshold(3);
+        headingDrivePid.setDerivativeThreshold(3.0 / 20);
+        
+        headingDrivePid.setEnableErrorThreshold(true);
+        headingDrivePid.setEnableDerivativeThreshold(true);
+    }
+
+    @Override
+    public void initialize() {
+        log.info("Initializing");
+    }
+
+    @Override
+    public void execute() {
+        double leftPower = leftJoystick.getVector().y;
+        double rightPower = leftJoystick.getVector().y;
+        
+        setTargetHeading();
+        
+        if(targetHeading != null && leftJoystick.getVector().getAngle() != poseSubsystem
+                                                                            .getCurrentFieldPose()
+                                                                            .getHeading()
+                                                                            .getValue()) {
+            leftPower =  poseSubsystem.getCurrentHeading().difference(targetHeading);
```

Comment:
you're setting the left power to the angle difference which probably isn't what you were intending?